src/views/ScheduledRestock.vue

```diff
@@ -0,0 +1,437 @@
+        'recurrenceTimeZone': this.userProfile.userTimeZone,
+        'tempExprId': job.jobStatus,
+        'statusId': "SERVICE_PENDING",
+        'runTimeEpoch': '',  // when updating a job clearning the epoch time, as job honors epoch time as runTime and the new job created also uses epoch time as runTime
+        'lastModifiedByUserLogin': this.userProfile.userLoginId
+      }
+
+      job?.runTime && (payload['runTime'] = job.runTime)
+      job?.sinceId && (payload['sinceId'] = job.sinceId)
+      job?.jobName && (payload['jobName'] = job.jobName)
+
+      try {
+        resp = await StockService.updateJob(payload)
+        if (!hasError(resp) && resp.data.successMessage) {
+          await this.store.dispatch('stock/fetchJobs')
+          showToast(translate('Service updated successfully'))
+        } else {
+          throw resp.data
+        }
+      } catch (err) {
+        showToast(translate('Failed to update job'))
+        logger.error(err)
+      }
+    },
+    mapFields(mapping) {
+      const fieldMapping = JSON.parse(JSON.stringify(mapping));
+
+      // TODO: Store an object in this.content variable, so everytime when accessing it, we don't need to use 0th index
+      const csvFields = Object.keys(this.content[0]);
+
+      const missingFields = Object.values(fieldMapping.value).filter(field => {
+        if(!csvFields.includes(field)) return field;
+      });
+
+      if(missingFields.length) showToast(translate("Some of the mapping fields are missing in the CSV: ", { missingFields: missingFields.join(", ") }))
+
+      Object.keys(fieldMapping.value).map((key) => {
+        if(!csvFields.includes(fieldMapping.value[key])){
+          fieldMapping.value[key] = "";
+        }
+      })
+      this.fieldMapping = fieldMapping.value;
+    },
+    async parse(event) {
+      const file = event.target.files[0];
+      try {
+        if (file) {
+          this.file = file;
+          this.content = await this.parseCsv(this.file);
+          this.fileColumns = Object.keys(this.content[0]);
+          showToast(translate("File uploaded successfully"));
+          this.selectedProductStoreId = this.productStores[0].productStoreId ? this.productStores[5].productStoreId : ''
+          this.updateProductStore(this.selectedProductStoreId)
+        } else {
+          showToast(translate("No new file upload. Please try again"));
+        }
+      } catch {
+        this.content = []
+        showToast(translate("Please upload a valid reset inventory csv to continue"));
+      }
+    },
+    async openScheduledRestockPopover(ev, job) {
+      const popover = await popoverController.create({
+        component: ScheduledRestockPopover,
+        event: ev,
+        translucent: true,
+        showBackdrop: false,
+        componentProps: { job }
+      });
+      return popover.present();
+    },
+    async review() {
+      const areAllFieldsSelected = Object.values(this.fieldMapping).every(field => field !== "");
```

Comment:
Add a loader as after clicking review it might take some time to move to the next page